README.md

```diff
@@ -1,90 +1,117 @@
 
-**What is a ULID?**
+## About
 
-ULIDs are Universally Unique Lexicographically Sortable Identifiers. This library adheres to [this specification](https://github.com/ulid/spec).
+### What is a ULID?
 
-**Why fork `ulidx`?** 
+ULIDs are universally Unique Lexicographically sortable IDentifiers. This library adheres to [the ULID specification](https://github.com/ulid/spec).
 
-See [this dissussion](https://github.com/perry-mitchell/ulidx/pull/6#issuecomment-1003190116). 
+### Monotonicity and ULID Time in Cloudflare Workers
 
-## Installation
+This library breaks slightly from tradition in that it generates `monotonic` `ULID`s by default. Normally `ULID`s are lexically sortable because the time component of the `ULID` is based on a high precision timestamp that is always moving forward. It is this timestamp that allows the `ULID` to be easily sorted in those cases.
 
-Install using npm by running: `npm install ulid-workers --save`.
+However, in the context of a Cloudflare Worker this assumption of always incrementing time breaks down a little. The Workers team has chosen, for security reasons, to freeze time within the context of a single worker request/response cycle.
 
-`ulid-workers` provides types and is written entirely in Typescript.
+The [Workers documentation](https://developers.cloudflare.com/workers/learning/security-model#step-1-disallow-timers-and-multi-threading) talks about the security reasons for this at length.
 
-## Usage
+> In Workers, you can get the current time using the JavaScript Date API, for example by calling `Date.now()`. However, the time value returned by this is not really the current time. Instead, it is the **time at which the network message was received** which caused the application to begin executing. While the application executes, time is locked in place.
 
-Import `ulid-workers` to generate new ULIDs:
+So if you were to generate multiple `ULID`s in a single request they would all have the same exact time component. In this case the normal behavior of other `ULID` libraries would be to have the random component be truly random. This means that if you generated 10 `ULID`s within a single request they would no longer be able to be guaranteed to be lexically sortable. Thus you lose one of the most important properties of the `ULID`.
 
-```typescript
-import { ulid } from "ulid-workers";
+The `ULID` spec defines a method for handling this situation. You can generate a [`monotonic`](https://github.com/ulid/spec#monotonicity) `ULID` which allows for the creation of more than `2^80` `ULID`s within a single millisecond while ensuring that they remain sortable. It does this by incrementing the random component by 1 bit in the least significant bit position (with carrying). Therefore the random component remains sortable even when the timestamp components are identical.
```

Comment:
The algorithm in the spec has two significant downsides. It produces easily predictable sequences of IDs, which may be a security concern. And by starting the random component at zero, two executions which happen to run in the same millisecond will create colliding ULIDs. [oklog/ulid provides monotonic entropy](https://godoc.org/github.com/oklog/ulid#Monotonic) parameterized in a way which addresses these issues. Please take a look, and maybe consider that approach instead.

Reply:
It does not start the random component at zero except from within the test suite. The examples in the README.md are using the zero filled randomess from the test just for clarity.

All randomness comes directly from `crypto.getRandomValues()`

Reply:
@peterbourgon If this PR is accepted and merged, you are free to create a PR that would show us an improved way to handle the generation of the random component as you have done on oklog/ulid.  I'd be curious to take a closer look.

Reply:
@grempe I will merge this PR, just testing it first. 

I will say, the zeroed out examples in the README threw me off for a minute until I read your tests and realized what you were doing. Perhaps the examples in the readme should show something more realistic.